File: backend/src/main/java/com/incidentcommand/dto/MetricsResponse.java
package com.incidentcommand.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MetricsResponse {
    private double meanTimeToResolveMinutes;
    private long totalIncidents;
    private long openIncidents;
    private long resolvedIncidents;
    private Map<String, Long> countBySeverity;
    private Map<String, Long> countByStatus;
    private Map<String, Long> countByTeam;
}

File: android/app/src/main/java/com/incidentcommand/android/data/remote/dto/MetricsResponse.kt
package com.incidentcommand.android.data.remote.dto

data class MetricsResponse(
    val meanTimeToResolveMinutes: Double,
    val totalIncidents: Long,
    val openIncidents: Long,
    val resolvedIncidents: Long,
    val countBySeverity: Map<String, Long>,
    val countByStatus: Map<String, Long>,
    val countByTeam: Map<String, Long>
)
